aspnetcore/signalr/Configuration.md

```diff
@@ -0,0 +1,124 @@
+---
+title: ASP.NET Core SignalR Configuration
+author: rachelappel
+description: Configure ASP.NET Core SignalR Apps
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: rachelap
+ms.date: 06/30/2018
+ms.prod: asp.net-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/configuration
+---
+
+# ASP.NET Core SignalR Configuration
+
+## [JSON]( https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Protocols.Json/JsonHubProtocolOptions.cs)/[MessagePack](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Protocols.MessagePack/MessagePackHubProtocolOptions.cs) serialization options
+
+MessagePack is a binary serialization format that provides serialization options for server and client. 
```

Comment:
We should expand this text a little bit. It doesn't really correlate with the code snippet. Also, don't link to the GitHub repository, that can easily change out from under the links. If we're going to link, we should link to API reference docs. How about:

----

## JSON/MessagePack Serialization Options

ASP.NET Core SignalR supports two protocols for encoding messages, JSON and MessagePack. Each protocol has configuration options that can be used to configure serialization.

### JSON Serialization Options

JSON serialization can be configured on both the Server and .NET Client by providing a delegate to the `AddJsonHubProtocol` method. The options object received by that delegate has a `PayloadSerializedSettings` which is a JSON.NET `JsonSerializerSettings` object that can be used to configure serialization of arguments and return values

On the server, add `AddJsonHubProtocol` to your `AddSignalR` call in `ConfigureServices`:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    // ... other services ...
    services.AddSignalR()
        .AddJsonHubProtocol(options => {
            // Configure the PayloadSerializerSettings here. 
            // For example, to use PascalCase property names instead of the default camelCase names:
            options.PayloadSerializerSettings.ContractResolver = new DefaultContractResolver();
            // See the JSON.NET Documentation for more details:
            // https://www.newtonsoft.com/json/help/html/Introduction.htm
        });
}
```

On the client, the same `AddJsonHubProtocol` extension method exists on `HubConnectionBuilder` (the `Microsoft.Extensions.DependencyInjection` namespace must be imported to see the method):

```csharp
// At the top of the file:
using Microsoft.Extensions.DependencyInjection;

// When constructing your connection:
var connection = new HubConnectionBuilder();
    .AddJsonHubProtocol(options => {
        // Configure the PayloadSerializerSettings here. 
        // For example, to use PascalCase property names instead of the default camelCase names:
        options.PayloadSerializerSettings.ContractResolver = new DefaultContractResolver();
        // See the JSON.NET Documentation for more details:
        // https://www.newtonsoft.com/json/help/html/Introduction.htm
    });
```

### MessagePack Serialization Options

MessagePack serialization can be configured by providing a delegate to the `AddMessagePackProtocol` call. See the documentation for MessagePack in SignalR [xref to https://github.com/aspnet/Docs/pull/6858 once it's merged] for more details.

Reply:
Also, it's not possible to configure JSON or MessagePack serialization on the JavaScript client at this time.